### PR TITLE
Closes #623 | Add/Update Dataloader MedEV

### LINKED ISSUE #623: Create dataset loader for MedEV
Dataloader name: `medev/medev.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?medev

|     Dataset| medev |
|-------------|---|
| Description | A high-quality Vietnamese-English parallel dataset constructed specifically for the medical domain, comprising approximately 360K sentence pairs |
| Subsets | - |
| Languages | vie |
| Tasks | Machine Translation |
| License | Unknown (unknown) |
| Homepage | https://huggingface.co/datasets/nhuvo/MedEV |
| HF URL | https://huggingface.co/datasets/nhuvo/MedEV |
| Paper URL | - |

### PR DESCRIPTION
**Pull request by patrickamadeus**
Closes #623 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/{my_dataset}/{my_dataset}.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py --subset_id {subset_name_without_source_or_seacrowd_suffix}`.
- [.] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### Tests
image

### CODE REVIEW

File: seacrowd/sea_datasets/medev/medev.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepath": data_dir["val"],
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        with open(filepath["en"], "r") as f:
```
Comment (elyanah-aco) [inline seacrowd/sea_datasets/medev/medev.py:136]: Let's set the encoding so this run on all systems - mine (Windows) doesn't run without it.
```suggestion
        with open(filepath["en"], "r", encoding="utf-8") as f:
```

File: seacrowd/sea_datasets/medev/medev.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            en_lines = f.readlines()
+        with open(filepath["vie"], "r") as f:
```
Comment (elyanah-aco) [inline seacrowd/sea_datasets/medev/medev.py:138]: Can you determine what encoding your computer uses for this file? Both `utf-8` and `latin-1` don't give me the intended result, and other encodings don't work at all.

File: seacrowd/sea_datasets/medev/medev.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            vie_lines = f.readlines()
+
+        if self.config.schema == "source":
+            for i, line in enumerate(en_lines + vie_lines):
+                yield i, {
+                    "text": line,
+                }
```
Comment (elyanah-aco) [inline seacrowd/sea_datasets/medev/medev.py:145]: ### 
```suggestion
                yield i, {
                    "id": str(i),
                    "text": line,
                }
```

File: seacrowd/sea_datasets/medev/medev.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+        elif self.config.schema == "seacrowd_t2t":
+            for i, (en_line, vie_line) in enumerate(list(zip(en_lines, vie_lines))):
+                yield i, {
+                    "id": str(i),
+                    "text_1": en_line,
+                    "text_2": vie_line,
+                    "text_1_name": "en",
```
Comment (elyanah-aco) [inline seacrowd/sea_datasets/medev/medev.py:153]: We use ISO 639-3 codes for languages.
```suggestion
                    "text_1_name": "eng",
```

**Review by elyanah-aco — COMMENTED** [review]
Thanks for the PR @patrickamadeus! Just suggesting some small changes:

File: seacrowd/sea_datasets/medev/medev.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                }
```
Comment (elyanah-aco) [inline seacrowd/sea_datasets/medev/medev.py:95]:
```suggestion
                {
                    "id": datasets.Value("string")
                    "text": datasets.Value("string"),
                }
```

**Review by akhdanfadh — COMMENTED** [review]
Thank you for the dataloader! Just add some minor edit and additional discussion.

File: seacrowd/sea_datasets/medev/medev.py (hunk 5 above)
Comment (patrickamadeus) [reply seacrowd/sea_datasets/medev/medev.py]: Hi! I'll address @akhdanfadh 's suggestion after everything concluded from ur guys' side!

File: seacrowd/sea_datasets/medev/medev.py (hunk 2 above)
Comment (patrickamadeus) [reply seacrowd/sea_datasets/medev/medev.py]: Mine is `utf-8`. it's working well from my side.

image

**Comment by patrickamadeus** [comment]
Hi @elyanah-aco ! I've addressed all of the suggestions! Appreciate the detailed review.

I will address suggestion from @akhdanfadh after your second opinion 🙏.

**Review by elyanah-aco — APPROVED** [review]
Thanks for the changes @patrickamadeus! OK now on my end (even the Vietnamese text)

**Comment by patrickamadeus** [comment]
Hi all @akhdanfadh @elyanah-aco ! The minor language expand is done! Thank you for all of the reviews. 🙏

**Comment by holylovenia** [comment]
Hi @akhdanfadh, I would like to let you know that we plan to finalize the calculation of the open contributions (e.g., dataloader implementations) **in 31 hours**, so it'd be great if we could wrap up the reviewing and merge this PR before then.

cc: @patrickamadeus

**Review by akhdanfadh — APPROVED** [review]
LGTM! Dataloader tested and worked. Merging in a bit.

**Pull request merged**